core-v1/src/main/scala/dsl/JustifiedBuildExprDsl.scala

```diff
@@ -0,0 +1,119 @@
+package com.rallyhealth.vapors.v1
+
+package dsl
+
+import algebra._
+import data.{FactTypeSet, Justified}
+import logic.Negation
+
+import cats.data.NonEmptyList
+import cats.{Foldable, Functor}
+
+trait JustifiedBuildExprDsl extends WrappedBuildExprDsl with JustifiedDslTypes {
+
+  override protected implicit final def compareWrapped: CompareWrapped[Justified] = CompareWrapped.justified
+
+  override protected implicit final def windowComparable: WindowComparable[Justified, OP] = WindowComparable.justified
+
+  override protected implicit final def extract: Extract[Justified] = Extract.justified
+
+  override protected implicit final def wrapConst: WrapConst[Justified] = WrapConst.justified
+
+  override protected def wrapElement[C[_], A](
+    outer: Justified[C[A]],
+    element: A,
+  ): Justified[A] =
+    Justified.byInference("elementOf", element, NonEmptyList.of(outer))
+
+  // TODO: Should this be visible outside this trait?
+  protected def dontShortCircuit: Boolean = false
```

Comment:
This is a question for future Jeff... I don't know whether short-circuiting will be useful for performance (and thus should be configurable per-DSL), or whether it will not be worth its weight.